packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -88,20 +102,28 @@ export const MatterportIntegration: React.FC = () => {
   const onMatterportModelIdChange = useCallback(
     (event) => {
       const value = event.detail.value;
-      if (value !== matterportModelId) {
-        setMatterportModelId(value);
+      if (value !== matterportModelId.current) {
+        matterportModelId.current = value;
         setDirty(true);
```

Comment:
i dont understand why we have all these dirty flags - react knows when to re-render based on updates to state and manages it efficiently. It feels like were trying to step outside of react and write imperative code

Reply:
Removed the dirty flag. As per the new changes, loss of focus from the input control will be used to update the value in scene file.